daemon/oci_linux.go

```diff
@@ -98,6 +100,31 @@ func setDevices(s *specs.Spec, c *container.Container) error {
 		for _, d := range hostDevices {
 			devs = append(devs, oci.Device(d))
 		}
+		for _, deviceMapping := range c.HostConfig.Devices {
+			if deviceMapping.PathOnHost == deviceMapping.PathInContainer {
+				logrus.WithField("container", c.ID).Warnf("Specifying a device %s is redundant for privileged mode", deviceMapping.PathOnHost)
```

Comment:
Looking at this again, I'm wondering if it's worth logging this, or if it's ok to just skip this warning: if there are no custom options (permissions) set, then the end result is exactly the same (with or without this device), so _perhaps_ it's ok to just ignore this without logging a warning